src/backends/treewalk/treewalk.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <structures/structures.hpp>
+
+#include <string>
+#include <variant>
+
+namespace lir::backend::treewalk {
+
+    using TreeWalkValue = std::variant<double, bool>;
+
+    // Simple tree walk interpreter, not complete
+    class TreeWalk {
+    public:
+
+        // Executes literal nodes
+        TreeWalkValue operator()(const lir::expressions::Literal& node) {
+            return { std::stod(node.value) };
+        }
+
+        // Executes unary nodes
+        TreeWalkValue operator()(const lir::expressions::Unary& node) {
+            auto expr = eval(node.expression);
+
+            switch (node.op) {
+                case Tokens::Op_Minus:
+                    return { -std::get<double>(expr) };
+                    
+                default:
+                    throw std::runtime_error("UNREACHABLE");
+            }
+        }
+
+        // Executes binary nodes
+        TreeWalkValue operator()(const lir::expressions::Binary& node) {
+            auto left  = eval(node.left);
+            auto right = eval(node.right);
+
+            switch (node.op) {
+                // Numerical operations
+                case Tokens::Op_Plus: 
+                    return { std::get<double>(left) + std::get<double>(right) };
+                case Tokens::Op_Minus:
+                    return { std::get<double>(left) - std::get<double>(right) };
```

Comment:
You could probably just do `std::get<double>` before the switch to stop code duplication since they are all double.